File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        
        k = 0 
        while True:
            k +=1 
            pre =k 
            mode =1 
            for n in nums:
                pre +=n 
                if pre <1:
                    mode =0 
                    break 
            if mode ==1:
                return k 

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
          int n = nums.size();
    int m = 200;
    
    int left = 1, right = m*n+1;
    
    while (left <right){
        int middle = (left +right) /2;
        int total = middle;
        bool isValid =true;
        
        for (int num : nums){
            total += num;
            
            
            if (total <1){
                isValid = false;
                break;
            }
        }
        if (isValid){
            right = middle;
        }
        else {
            left = middle +1;
        }
    
    }

    return left;
    }
};